devfiles/java-gradle/devfile.yaml

```diff
@@ -20,7 +20,7 @@ components:
     args: ['infinity']
     env:
       - name: GRADLE_USER_HOME
-        value: /home/user/.gradle
+        value: /home/gradle/.gradle
```

Comment:
just a question: do we need two homes? or 
```
      - name: HOME
        value: /home/user
```
Should be changed to 
```
      - name: HOME
        value: /home/gradle
```
?

Reply:
I set to `HOME` to `/home/user` originally because that's how legacy Che workspaces worked and there were still some environment variables injected by wsmaster that assumed that. I don't know if that's still the case.

`/home/gradle/.gradle` is persisted as a volume right? data should not be wiped out. 

I am afraid that setting `/projects/` as `HOME` may have unexpected side effects. We may want to set the [k8s container workingDir](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#container-v1-core) instead.

Reply:
@l0rd That makes sense to me but I'm still not sure the best way to proceed here. While `/home/gradle/.gradle` is persisted, `/home/gradle` is not; are we okay with this (e.g. it would prevent using a persistent `.bashrc`, etc.). This issue is probably another, larger discussion, though. 

I've updated `$HOME` to be `/home/gradle`. What is currently happening, AFAICT, is that whatever directory we're calling `$HOME` is being created at container start up, so I don't think this will cause issues.

Reply:
We are ok not to persist full `$HOME`. This is a cool idea (a user would find his dotfiles in every workspace) but I don't think there has ever been a requirement for that.

Reply:
Anyway I am +1 to merge this PR as it is